blocks/bricks/conv.py

```diff
@@ -147,14 +147,33 @@ class ConvolutionalLayer(Sequence, Initializable):
     """
     def __init__(self, filter_size, num_filters, num_channels, pooling_size,
                  activation, conv_step=(1, 1), pooling_step=None,
-                 border_mode='valid', **kwargs):
-        convolution = Convolutional(filter_size, num_filters,
-                                    num_channels, conv_step, border_mode)
-        pooling = MaxPooling(pooling_size, pooling_step)
+                 border_mode='valid', input_dim=None, **kwargs):
+        self.input_dim = input_dim
+        self.convolution = Convolutional(filter_size, num_filters,
+                                         num_channels, conv_step, border_mode)
+        self.pooling = MaxPooling(pooling_size, pooling_step)
         super(ConvolutionalLayer, self).__init__(
-            application_methods=[convolution.apply, activation, pooling.apply],
+            application_methods=[self.convolution.apply, activation,
+                                 self.pooling.apply],
             **kwargs)
 
+    def get_dim(self, name):
+        if name == 'input_':
+            return self.input_dim
+        if name == 'output':
+            if hasattr(self, '_output_dim'):
```

Comment:
I think it's okay to repeat this computation, no need to store it in a private variable.